CleanWeather/Networking/WeatherNetworking.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  WeatherNetworking.swift
+//  CleanWeather
+//
+//  Created by Marek Skrzelowski on 12/11/2019.
+//  Copyright © 2019 vandermesis. All rights reserved.
+//
+
+import Foundation
+
+typealias FetchWeatherCompletion = ([Weather]) -> Void
+
+protocol WeatherNetworking {
+    func fetchWeather(completion: FetchWeatherCompletion?)
+}
+
+final class WeatherNetworkingImpl: WeatherNetworking {
+    
+    func fetchWeather(completion: FetchWeatherCompletion?) {
```

Comment:
może nazwa ktora wiecej mowi? fetchTodaysWeatherForAllCities

Reply:
Jeśli chodzi o tą metodę to właśnie zastanawiam się nad tym jak to powinno być rozwiązane.
Czy robiąc zapytanie do API nie lepiej ściągnąć na raz wszystkie dane? Tzn: Current, Hourly i Days? I potem z tego modelu przekazać je do odpowiednich scen?
Tak to miałem rozwiązane w swojej aplikacji?
Czy powinno być tak, że są osobne modele to Current, Hourly i Days i jeszcze Past?
W mojej koncepcji były dwa zapytania do API:
1. Dane Current, Hours, Days (temperatura, ikona, nazwa miasta), Dane godzinowe dla dnia (godzina, temperatura max, min, ikona, prawdopodobieństwo opadu), Dane na kilka dni do przodu (dzień, temperatura max, min, ikona, opad)
2. Dane Past (temperatura, ikona, nazwa miasta) - zapytanie o dane historyczne dla konkretnego miasta
Czy może to nie tak powinno być?
Robimy zapytanie i zapisujemy je do modelu za każdym razem jak użytkownik przechodzi do kolejnej sceny? W tej drugiej strategii będzie to generowało bardzo dużo zapytań, bo użytkownik może skakać z ekranu na ekran. Co za tym idzie ewentualny koszt utrzymania aplikacji będzie większy. Większa liczba zapytań użytkowników = większa cena za wykorzystanie API.

Reply:
Wszystko zalezy od zalozen i tego co daje api, co chce klient, a ze ty jestes jednoczesnie klientem i developerm to decydujesz jak uważasz, w praktyce częściej chyba jednak robimy mniejsze zapytania ale ich więcej, żeby też mioeć pewność ze dane są aktualne

Reply:
Ok. Rozumiem, że w takim razie usuwamy z modelu CityWeather właściwości potrzebne do innch modułów: np CityDetails tak?
Chodzi mi o:
`let hourlyTempMin: [Double]
    let hourlyTempMax: [Double]
    let dailyTempMin: [Double]
    let dailyTempMax: [Double]`

Reply:
Narazie nie mamy widoku wiec niie wiem co chcesz na nim wyswietlic, zostaw i zobaczymy jak wyjdzie później